### PR TITLE
Add Chapter 4 exercise 1 the sum of a range  KAN-24

### PR DESCRIPTION
**Pull request by insert-oreyes**
- chapter4: setup ready
- json: db updte with ch4 information
- Exercise 1 working

https://github.com/jon-oscar/eloquent-js/assets/99765175/586a1714-f1c5-4fa9-8a07-9d0e830324b6

### CODE REVIEW

File: pages/chapter4.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                  title={devOscar[2].title}
+                  details={devOscar[2].details}
+                  code={() => <Alist />}
+                />
+                <PageCard
+                  id={devOscar[3].id}
+                  title={devOscar[3].title}
+                  details={devOscar[3].details}
+                  code={() => <DeepComparison />}
+                />
```
Comment (jportella93) [inline pages/chapter4.tsx]: This shouldn't be here, this PR is only for the `SumOfARange` exercise. Please remove this and all the files for the other exercises from this PR.

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
+import Image from 'next/image';
+import React, { useState } from 'react';
+
+/**
+ * Calculates the sum of a range of numbers.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @returns An array containing all the numbers in the range.
+ */
+export const calcRangeSum = (start: number, end: number) => {
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Can you add a type for what's returned from this function? https://www.phind.com/search?cache=fsymi8ktusn4n10wtrfsz214

Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: Also the naming and docs for this function are wrong, it doesn't do any summing or calculation. A better name would be `createRange` or `getRange`

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+};
+
+/**
+ * Calculates the sum of a range of numbers based on a given start, end, and step.
+ * @param start - The starting number of the range.
+ * @param end - The ending number of the range.
+ * @param step - The step to increment or decrement the range by.
+ * @returns An array of numbers representing the range.
+ */
+export const calcRangeSumMod = (start: number, end: number, step: number) => {
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Can you type the return value?

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  }
+  return range;
+};
+
+/**
+ * Calculates the sum of all numbers in an array.
+ * @param arr - The array of numbers to sum.
+ * @returns The sum of all numbers in the array.
+ */
+export const arraySum = (arr: number[]) => {
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Type return value

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+ * @returns An array of numbers representing the range.
+ */
+export const calcRangeSumMod = (start: number, end: number, step: number) => {
+  const range = [];
+
+  if (step === 1) {
+    // if step is 1, then we can just use the original function
+    for (let i = start; i <= end; i += step) {
+      range.push(i);
+    }
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Then just reuse it instead of copy pasting it
```js
range = calcRangeSum(start, end)
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+  } else if (step < 1) {
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: I think you want to check for a negative step, then you should do `step < 0`

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    // if step is less than 1, then we need to decrement the range
+    for (let i = start; i >= end; i += step) {
+      range.push(i);
+    }
+  } else {
+    return;
+  }
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: So it doesn't work when `step > 1`? 
Another thought: What should happen when step is 0?

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+ * @returns The sum of all numbers in the array.
+ */
+export const arraySum = (arr: number[]) => {
+  let sum = 0;
+  for (let i = 0; i < arr.length; i++) sum += arr[i];
+  return sum;
+};
+
+const animals = [
+  { id: 1, name: 'Bear', img: '/bear.png' },
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx:9]: Why do we need `id`s? Isn't the name unique already, so it can be used as id?

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+ * Calculates the sum of all numbers in an array.
+ * @param arr - The array of numbers to sum.
+ * @returns The sum of all numbers in the array.
+ */
+export const arraySum = (arr: number[]) => {
+  let sum = 0;
+  for (let i = 0; i < arr.length; i++) sum += arr[i];
+  return sum;
+};
+
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: This file is already getting long and hard to reason about, I suggest you refactor this, moving the helper functions above to their own files, with their own test files. Here's a quick way to do it https://www.youtube.com/watch?v=uito4KkQ3Sk

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+];
+
+const initialAnimals = {
+  start: 1,
+  end: 10,
+  step: 1,
+};
+
+export default function SumOfARange() {
+  const [showAnimals, setShowAnimals] = useState(initialAnimals);
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: This aren't animals, it is a range (used to render animals)
```js
const [range, setRange] = useState(initialRange);
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+  { id: 4, name: 'Monkey', img: '/monkey.png' },
+  { id: 5, name: 'Elephant', img: '/elephant.png' },
+  { id: 6, name: 'Hippo', img: '/hippo.png' },
+  { id: 7, name: 'Lion', img: '/lion.png' },
+  { id: 8, name: 'Lynx', img: '/lynx.png' },
+  { id: 9, name: 'Ostrich', img: '/ostrich.png' },
+  { id: 10, name: 'Wolf', img: '/wolf.png' },
+];
+
+const initialAnimals = {
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: This aren't animals, it is a range
```js
const initialRange = {
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+};
+
+export default function SumOfARange() {
+  const [showAnimals, setShowAnimals] = useState(initialAnimals);
+
+  /**
+   * Handles the change event of the end select element and updates the showAnimals state accordingly.
+   * @param event - The change event of the start select element.
+   */
+
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: extra whiteline

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+  start: 1,
+  end: 10,
+  step: 1,
+};
+
+export default function SumOfARange() {
+  const [showAnimals, setShowAnimals] = useState(initialAnimals);
+
+  /**
+   * Handles the change event of the end select element and updates the showAnimals state accordingly.
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: end element or start element?

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+          <p className='mt-3 mb-2'>Sum of animals</p>
+
+          <p className='flex flex-row justify-start py-1'>
+            {arraySum(calcRangeSum(showAnimals.start, showAnimals.end))}
+          </p>
+        </div>
+      </div>
+    </div>
+  );
+}
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx:155]: I've been playing with the exercise on the live preview and it is a bit hard to understand what is going on. I think it would be clearer by:
1. Setting an exercise description, explaining what is going on
2. Adding the value of each animal next to their name, so the sum makes sense to the viewer

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+      if (showAnimals.start > showAnimals.end) {
+        showAnimals.step = -1;
+      }
+      return showAnimals;
+    });
+  }
+
+  return (
+    <div>
+      <div className='animals container grid grid-cols-5 grid-rows-2 gap-3'>
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: what does this `animals` classname do?

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+        {calcRangeSumMod(
+          showAnimals.start,
+          showAnimals.end,
+          showAnimals.step
+        )?.map((num) => {
+          if (num === animals[num - 1].id) {
+            return (
+              <div
+                className='w-24 h-24 flex flex-col text-center justify-center items-center rounded-md space-y-1'
+                key={animals[num - 1].id}
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Just use the name, it is unique
```js
key={animals[num - 1].name}
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+              >
+                <Image
+                  width={60}
+                  height={60}
+                  src={animals[num - 1].img}
+                  alt={animals[num - 1].name}
+                />
+                <div>{animals[num - 1].name}</div>
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: A more semantic HTML approach for image + description is `figure` https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 18]
```diff
@@ -0,0 +1,211 @@
...
+    <div>
+      <div className='animals container grid grid-cols-5 grid-rows-2 gap-3'>
+        {calcRangeSumMod(
+          showAnimals.start,
+          showAnimals.end,
+          showAnimals.step
+        )?.map((num) => {
+          if (num === animals[num - 1].id) {
+            return (
+              <div
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: In JSX, returning `false` doesn't render anything on screen so you can actually do this shorter version instead
```js
** {calcRangeSumMod(
          showAnimals.start,
          showAnimals.end,
          showAnimals.step
        )?.map((num) => num === animals[num - 1].id && <div
```

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+    expect(calcRangeSum(-2, 2)).toEqual([-2, -1, 0, 1, 2]);
+    expect(calcRangeSum(0, 0)).toEqual([0]);
+  });
+});
+describe('calcRangeSumMod', () => {
+  it('returns an array containing all the numbers in the range', () => {
+    expect(calcRangeSumMod(1, 5, 1)).toEqual([1, 2, 3, 4, 5]);
+    expect(calcRangeSumMod(5, 1, -1)).toEqual([5, 4, 3, 2, 1]);
+    expect(calcRangeSumMod(1, 5, 0)).toEqual([]);
+  });
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.test.tsx]: Can we add a test for step > 1?

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+describe('arraySum', () => {
+  it('returns the sum of all elements in the array', () => {
+    expect(arraySum([1, 2, 3, 4, 5])).toEqual(15);
+    expect(arraySum([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])).toEqual(55);
+  });
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.test.tsx]: Can we add a test for negative numbers?

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+describe('SumOfARange', () => {
+  it('renders the animal images based on the selected range', () => {
+    render(<SumOfARange />);
+    const bearImage = screen.getByAltText('Bear');
+    expect(bearImage).toBeInTheDocument();
+    const wolfImage = screen.getByAltText('Wolf');
+    expect(wolfImage).toBeInTheDocument();
+  });
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.test.tsx:27]: This is only testing the initial state. Can we test the expected animals are rendered when selecting different ranges?

**Review by jportella93 — COMMENTED** [review]
Left some comments

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 3 above)
Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: Also the naming and docs for this function are wrong, it doesn't do any summing or calculation. A better name would be `createRangeWithStep` or `getRangeWithStep`

Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: Actually this is not exactly what the exercise requests
>As a bonus assignment, **modify** your range function to take an **optional third argument** that indicates the “step” value used when building the array. **If no step is given**, the elements go up by increments of one, corresponding to the **old behavior**. 

You should modify the `createRange` function to work with an optional third parameter, not create a new function for this

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx (hunk 19 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.test.tsx]: test added and passed, I had to modify the conditional parameter from === 1 to <= 1

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx (hunk 20 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.test.tsx]: test added and passed

File: components/exercises/chapter-4-oscar/SumOfARange.test.tsx (hunk 21 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.test.tsx:27]: test renders animals when selecting different ranges added and passed

File: pages/chapter4.tsx (hunk 1 above)
Comment (insert-oreyes) [reply pages/chapter4.tsx]: files removed

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 2 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: Fixed, thank you, very interesting the phind link

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 3 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: I believed it would be a good practice for me to keep the start and the modification so I could see the difference. I just removed it and modified the file according to your comments

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 5 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored, no need to add this

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 6 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored, no need to add this

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 7 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored, I added an error in case the step is 0

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 8 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx:9]: I use it to match the getRangeWithStep with the animal in the state

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 10 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 11 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 12 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 13 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 15 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: it didn't mean anything I needed a way to identify the containers while I was designing them, I should have removed it

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 16 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 17 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 18 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: reformat, thank you

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 4 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: included, thank you

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 9 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: I exported the functions

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 12 above)
Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: there's still an extra whiteline here, can you remove it?

Generally, function docs should be right above the function they describe, without any line breaks

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+ * @property {number} step - The step between each number in the range.
+ */
+const initialRange = {
+  start: 1,
+  end: 10,
+  step: 1,
+};
+
+export default function SumOfARange(): React.JSX.Element {
+  const [range, setRange] = useState(initialRange);
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx:35]: Missing white line between line 36 and 37

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 17 above)
Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: `figure` requires the text of the image to be a `figcaption` element, check the link again

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+                  <option value='1'>1</option>
+                  <option value='2'>2</option>
+                  <option value='3'>3</option>
+                  <option value='4'>4</option>
+                  <option value='5'>5</option>
+                  <option value='6'>6</option>
+                  <option value='7'>7</option>
+                  <option value='8'>8</option>
+                  <option value='9'>9</option>
+                  <option value='10'>10</option>
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: This repetition could be replaced by array.map

File: components/exercises/chapter-4-oscar/SumOfARange.tsx [hunk 24]
```diff
@@ -0,0 +1,167 @@
...
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/SumOfARange.tsx]: Ditto

File: constants/chapter.json [hunk 25]
```diff
@@ -86,12 +86,52 @@
...
-    "exercises": "",
+    "exercises": "The sum of a range, reversing an array, a list, deep comparison",
     "link": "/chapter4",
-    "details": ""
+    "details": "Objects and arrays (which are a specific kind of object) provide ways to group several values into a single value. Conceptually, this allows us to put a bunch of related things in a bag and run around with the bag, instead of wrapping our arms around all of the individual things and trying to hold on to them separately.",
+    "devOscar": [
+      {
+        "id": 1,
+        "title": "The sum of a range",
+        "details": "The dropdown option adjust the quantity of animals to be shown. Sum of animals, is a function that takes an array and returns the sum of these numbers."
```
Comment (jportella93) [inline constants/chapter.json]: What about:
```js
"details": "Adjust the range of animals to be shown through the dropdown selectors. Each animal represents a value from 1 to 10. Sum of animals represents the sum of the selected animals."
```

File: components/exercises/chapter-4-oscar/getRangeWithStep.tsx [hunk 26]
```diff
@@ -0,0 +1,19 @@
+export const getRangeWithStep = (
```
Comment (jportella93) [inline components/exercises/chapter-4-oscar/getRangeWithStep.tsx]: Missing docstring

File: pageTests/Chapter4.test.tsx [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+      details.startsWith(
+        'Objects and arrays (which are a specific kind of object) provide ways to group several'
+      )
+    );
+    expect(title).toBeInTheDocument();
+    expect(subtitle).toBeInTheDocument();
+    expect(details).toBeInTheDocument();
+  });
+
+  it('renders the The sum of a range, Reversing an array, A list and Deep comparison exercises', () => {
```
Comment (jportella93) [inline pageTests/Chapter4.test.tsx]: not true yet
```js
it('renders the sum of a range exercise', () => {
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 23 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 24 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored

File: constants/chapter.json (hunk 25 above)
Comment (insert-oreyes) [reply constants/chapter.json]: refactored, thank you

File: components/exercises/chapter-4-oscar/getRangeWithStep.tsx (hunk 26 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/getRangeWithStep.tsx]: added

File: pageTests/Chapter4.test.tsx (hunk 27 above)
Comment (insert-oreyes) [reply pageTests/Chapter4.test.tsx]: revised

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 12 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: fixed

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 17 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: figcaption tag added

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 22 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx:35]: line removed

Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx:35]: still missing a whiteline. It should be like this
```js
const [range, setRange] = useState(initialRange);

  /**
   * Handles the change event of the start select element and updates the range state accordingly.
```

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 12 above)
Comment (jportella93) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: There's still an extra whiteline. It should be like this:
```js
   * @param event - The change event of the start select element.
   */
   function handleChangeStart(event: React.ChangeEvent<HTMLSelectElement>) {
```

**Review by jportella93 — APPROVED** [review]
Just two comments to address, otherwise looks good to me. Approving to unblock.

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 12 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx]: refactored, thank you

File: components/exercises/chapter-4-oscar/SumOfARange.tsx (hunk 22 above)
Comment (insert-oreyes) [reply components/exercises/chapter-4-oscar/SumOfARange.tsx:35]: refactored
